docs/src/operations/upgrade.md

```diff
@@ -0,0 +1,152 @@
+1. Make a backup of the `data` folder
+
+#### From broker version >= 24.4
+
+The [Partitions Admin Endpoint](#partitions-admin-endpoint) can be used to query the status of the partitions and perform operations for the upgrade.
+
+1. Stop the workflow processing and trigger a snapshot creation
+    * Call the `prepareUpgrade` operation on the partitions endpoint
+    * Query the status of the partitions until the following conditions are met for all partitions:
+        * `processedPositionInSnapshot` is equal to `processedPosition`
+        * all followers (role: `FOLLOWER`) have the same `snapshotId` as the leader (role: `LEADER`)
+1. Make a backup of the `data` folder
+
+### Performing the Upgrade
+
+1. Shut down the broker
+1. Replace the `/bin` and `/lib` folders with the versions of the new distribution
+1. Start up the broker
+
+### Verifying the Upgrade
+
+The upgrade is successful if the following conditions are met:
+
+* the broker is ready (see [Ready Check](/operations/health.md#ready-check))
+* the broker is healthy (see [Health Check](/operations/health.md#health-check))
+* all partitions are healthy (see the [Metric](/operations/metrics.md#metrics-related-to-health) `zeebe_health`)
+* the stream processors of the partition leaders are in the phase `PROCESSING` (see [Partitions Admin Endpoint](#partitions-admin-endpoint))
+
+If the upgrade failed because of a known issue then a partition change its status to unhealthy, and the log output may contain the following error message:
+
+<details>
+  <summary>Sample Upgrade Error Message</summary>
+  <p>
+
+```
+Unexpected error on recovery happens.
+io.zeebe.engine.processor.InconsistentReprocessingException: Reprocessing issue detected!
+  Restore the data from a backup and follow the recommended upgrade procedure. [cause:
+  "The key of the record on the log stream doesn't match to the record from reprocessing.",
+  log-stream-record: {"partitionId":1,"value":{"version":1,"bpmnProcessId":"parallel-tasks",
+  "workflowKey":2251799813685249,"parentElementInstanceKey":-1,"parentWorkflowInstanceKey":-1,
+  "bpmnElementType":"PARALLEL_GATEWAY","flowScopeKey":2251799813685251,
+  "elementId":"ExclusiveGateway_0tkgnd5","workflowInstanceKey":2251799813685251},
+  "key":2251799813685256,"sourceRecordPosition":4294997784,"valueType":"WORKFLOW_INSTANCE",
+  "timestamp":1601025180728,"recordType":"EVENT","intent":"ELEMENT_ACTIVATING",
+  "rejectionType":"NULL_VAL","rejectionReason":"","position":4294998112},
+  reprocessing-record: {key=2251799813685255, sourceRecordPosition=4294997784,
+  intent=WorkflowInstanceIntent:ELEMENT_ACTIVATING, recordType=EVENT}]
+```
+
+  </p>
+</details>
+
+In this case, the broker should be rolled back to the previous version and the backup should be restored. Ensure that the upgrade was prepared correctly. If it is still unclear why it was not successful then please contact the Zeebe team and ask for guidance.
+
+
+## Rolling Upgrade
```

Comment:
Should we move this section to the beginning? Mention that this is usual way to upgrade - but because of the current issue follow the following upgrade procedure temporarily.